Source/DafnyLanguageServer/Workspace/CompilationManager.cs

```diff
@@ -114,6 +113,10 @@ public class CompilationManager {
       var resolvedDocument = await ResolvedDocument;
       var translatedDocument = await PrepareVerificationTasksAsync(resolvedDocument, cancellationSource.Token);
       documentUpdates.OnNext(translatedDocument);
+      foreach (var task in translatedDocument.VerificationTasks!) {
+        cancellationSource.Token.Register(task.Cancel);
```

Comment:
Oh !! Is this where you are resolving the fact that tasks were not cancelled?

Reply:
Jep, pretty simple.